cpp/src/arrow/util/bit-util.h

```diff
@@ -329,6 +327,30 @@ static inline void SetBitTo(uint8_t* bits, int64_t i, bool bit_is_set) {
                  kBitmask[i % 8];
 }
 
+static inline void SetBitsTo(uint8_t* bits, int64_t i_begin, int64_t i_end,
+                             bool bits_are_set) {
+  if (i_begin == i_end) return;
+
+  auto bytes_begin = i_begin / 8;
+  auto bytes_end = CeilDiv(i_end, 8);
+
+  auto first_byte = bits[bytes_begin];
+  auto last_byte = bits[bytes_end - 1];
+
+  std::memset(bits + bytes_begin, bits_are_set ? 0xFF : 0x00,
+              static_cast<size_t>(bytes_end - bytes_begin));
+
+  auto first_byte_mask = kPrecedingBitmask[i_begin % 8];
```

Comment:
Ideally I think we would like to treat the first byte independently before the `memset`, in case the specified length is larger than the CPU cache.